### PR TITLE
fix(extractor): avoid extracting nested tokens as color-palette

### LINKED ISSUE #769: Invalid CSS when extending theme with semanticTokens (colors and shadows)
### Description

This is related to the closed ticket #754, I'm using `base` and `_dark` keys to define the colors, however I got the same generated css variables 

```css
--colors-color-palette-0: var(--colors-color-palette-0);
  --colors-color-palette-10: var(--colors-color-palette-10);
  --colors-color-palette-20: var(--colors-color-palette-20);
  --colors-color-palette-30: var(--colors-color-palette-30);
  --colors-color-palette-40: var(--colors-color-palette-40);
  --colors-color-palette-50: var(--colors-color-palette-50);
  --colors-color-palette-60: var(--colors-color-palette-60);
  --colors-color-palette-70: var(--colors-color-palette-70);
  --colors-color-palette-80: var(--colors-color-palette-80);
  --colors-color-palette-90: var(--colors-color-palette-90);
  --colors-color-palette-95: var(--colors-color-palette-95);
  --colors-color-palette-99: var(--colors-color-palette-99);
  --colors-color-palette-100: var(--colors-color-palette-100);
...
```

one more thing I noticed, when using the semantic shadows with the array style, it separates the keys into multiple css vars 
for example here is the shadow token 

```ts
  e1: {
    value: {
      base: ['0px 1px 2px rgba(0, 0, 0, 0.3)', '0px 1px 3px 1px rgba(0, 0, 0, 0.15)'],
      _dark: ['0px 1px 3px 1px rgba(0, 0, 0, 0.15)', '0px 1px 2px rgba(0, 0, 0, 0.3)'],
    },
  },

```

and here is the generated css 

```css
 :where(:root, :host) {
  --shadows-e1: 0px 1px 2px rgba(0, 0, 0, 0.3), 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
  ...
 }

{
  --shadows-e1: 0px 1px 2px rgba(0, 0, 0, 0.3);
}

{
  --shadows-e1: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
}

.dark {
  --shadows-e1: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
}

.dark {
  --shadows-e1: 0px 1px 2px rgba(0, 0, 0, 0.3);
}
```

one more thing I just noticed inside the generated `index.css`, that the last line of all generated css vars are missing a semicolon

 

### Link to Reproduction

https://codesandbox.io/p/github/omarkhatibco/pandacss-test/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522clj3g2y4k000b3b6nca8rhglx%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522clj3g2y4k000d3b6npw8t91vd%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clj3g2y4k000b3b6nca8rhglx%2522%253A%257B%2522id%2522%253A%2522clj3g2y4k000b3b6nca8rhglx%2522%252C%2522activeTabId%2522%253A%2522clj5gofe700dy3b6nu4ni2f05%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clj3g2y4k000a3b6ne53qyvb8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.node.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fstyled-system%252Ftokens%252Findex.css%2522%252C%2522id%2522%253A%2522clj5gofe700dy3b6nu4ni2f05%2522%252C%2522mode%2522%253A%2522temporary%2522%257D%255D%257D%252C%2522clj3g2y4k000d3b6npw8t91vd%2522%253A%257B%2522id%2522%253A%2522clj3g2y4k000d3b6npw8t91vd%2522%252C%2522activeTabId%2522%253A%2522clj3g55w900na3b6n5jyzb1zg%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clj3g527g00h73b6nc0l440p1%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clj3g55w900na3b6n5jyzb1zg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522PROJECT_SETUP%2522%252C%2522id%2522%253A%2522clj4bqh0w00693b6on1jj7eri%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to reproduce

Check  generated `index.ts` inside token folder

### JS Framework

React (TS)

### Panda CSS Version

0.3.2

### Browser

latest

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by isBatak**
Closes #769

## 📝 Description

This PR will:
- [x] avoid extracting nested tokens as `color-palette` css variables
- [x] avoid extracting shadow array as a separate unnamed block for the custom dark condition

## ⛳️ Current behavior (updates)

Described in the issue report

## 🚀 New behavior

This fix will not extract color palette tokens as the css css variables and correctly extract array tokens (for example shadow).

## 💣 Is this a breaking change (Yes/No):

Should not be a breaking change

### CODE REVIEW

File: packages/studio/styled-system/css/css.d.ts [hunk 1]
```diff
@@ -1,9 +1,3 @@
...
 import type { SystemStyleObject } from '../types'
-
-interface CssFunction {
-  (styles: SystemStyleObject): string
-  raw: (styles: SystemStyleObject) => SystemStyleObject
-}
-
-export declare const css: CssFunction;
\ No newline at end of file
+export declare function css(styles: SystemStyleObject): string
```
Comment (astahmer) [inline packages/studio/styled-system/css/css.d.ts:3]: you probably need to `git rebase origin/main`

File: packages/studio/styled-system/jsx/is-valid-prop.mjs [hunk 2]
```diff
@@ -436,7 +436,7 @@ var userGenerated = [
   '_light',
   '_osDark',
   '_osLight',
-  '_highContrast',
+  '_highConstrast',
```
Comment (isBatak) [inline packages/studio/styled-system/jsx/is-valid-prop.mjs:439]: idk what happened here... I didn't change this

**Review by astahmer — APPROVED** [review]

File: packages/studio/styled-system/jsx/is-valid-prop.mjs (hunk 2 above)
Comment (astahmer) [reply packages/studio/styled-system/jsx/is-valid-prop.mjs:439]: this is a change from a few days ago, this should have been updated with it, it's fine

File: packages/studio/styled-system/css/css.d.ts (hunk 1 above)
Comment (isBatak) [reply packages/studio/styled-system/css/css.d.ts]: Yeah I did that and still... something weird happened here

**Pull request merged**